frontend/leaderboard/src/Challenges.re

```diff
@@ -119,22 +119,21 @@ let highestSnarkFeeCollected = blocks => {
   );
 };
 
-let transactionsSentToAddress = blocks => {
-  Array.fold_left(
-    (map, block: Types.NewBlock.t) => {
-      StringMap.update(
-        block.data.newBlock.transactions.feeTransfer.recipient,
-        value =>
-          switch (value) {
-          | Some(transactionCount) => Some(transactionCount + 1)
-          | None => Some(1)
-          },
-        map,
-      )
-    },
-    StringMap.empty,
-    blocks,
-  );
+// Loop through the transactions in each block
+// fold left and increment map with the public key and when a transaction's toAccount is the same as address
+let calculateTransactionsSentToAddress =
+    (map, block: Types.NewBlock.data, address) => {
```

Comment:
I would expect having a public key string as the address would be the most convenient.
Why do we need to pass in a map to this function, and why does it accept only a single block?

Reply:
i see, does that mean we'd just check for `userCommand.toAccount.publicKey === address` instead? or should i add a type signature in the function `address: publicKey`? and oops - it should be using a similar pattern to what martin had above and calculate the property over all blocks - i will change!